spec/javascripts/app/views/location_stream_spec.js

```diff
@@ -0,0 +1,39 @@
+describe("app.views.LocationMap", function() {
+  beforeEach(function(){
+    this.post = factory.post();
+    this.view = new app.views.LocationStream({model : this.post});
+  });
+
+  describe("toggleMap", function() {
+    context("with location provided", function() {
+      beforeEach(function(){
+        this.post.set({location : factory.location()}); // set location
+        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
+        this.view.render();
+      });
+
+      it("should contain a map container", function() {
+        expect(this.view.$el[0]).toContainElement(".mapContainer");
+      });
+
+      it("should initialize map", function() {
+        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        this.view.toggleMap();
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+
+      it("should change display status on every click", function() {
```

Comment:
> Btw, how did you find that solution?

First I found out (with some `console.log(...)`) that the toggle function really has been called but didn't do anything. Then I called the toggle function manually in the spec and still saw no change so I wanted to get a minimal failing spec and added some div to `spec.content()` and called `toggle()` on that div but that worked, so I guessed that one would have to add the element you want to change to the page that jasmine loads in the browser.

> Can we somewhere read anything about that spec.content construction?

https://github.com/diaspora/diaspora/blob/develop/spec/javascripts/helpers/SpecHelper.js#L149